packages/slice-machine/src/modules/pushChangesSaga/index.ts

```diff
@@ -136,6 +143,18 @@ export function* changesPushSaga({
       })
     );
   } catch (error) {
+    // if (isUnauthenticatedError(error) || isUnauthorizedError(error)) { // TODO: This should work
+    const isSMError = (error: unknown): error is SliceMachineError =>
+      typeof error === "object" && error !== null;
+
+    if (
+      isSMError(error) &&
+      (error.name === "UnauthenticatedError" ||
+        error.name === "UnauthorizedError")
+    ) {
+      yield put(modalOpenCreator({ modalKey: ModalKeysEnum.LOGIN }));
+      return;
+    }
```

Comment:
I shared with you on Slack another way to do that, I think it'll make it more consistent than doing it across the Slice Machine UI codebase while we wait for a better fix on error serialization.